services/consensuscontext/service.go

```diff
@@ -31,6 +41,10 @@ func NewConsensusContext(
 		stateStorage:    stateStorage,
 		config:          config,
 		logger:          logger.WithTags(LogTag),
+		metrics: metrics{
```

Comment:
It feels a bit wrong to initialize it here, maybe we need a factory method per service like `consensuscontext.NewMetrics()`?

Reply:
why?